tests/test_validators.py

```diff
@@ -1256,3 +1256,35 @@ def validate_foo(cls, v):
     with pytest.raises(RuntimeError, match='test error'):
         model.foo = 'raise_exception'
     assert model.foo == 'foo'
+
+
+def test_bool_validator():
```

Comment:
A lot of those tests are already in parametrized test `test_types.py::test_default_validators`
You can remove all of this and simply add some usecases there.